go-controller/pkg/ovn/kubevirt.go

```diff
@@ -46,18 +53,261 @@ func (oc *DefaultNetworkController) deleteDHCPOptions(pod *kapi.Pod) error {
+	if err != nil {
+		return fmt.Errorf("failed configuring remote pod routing when getting switch current and original name: %v", err)
+	}
+
+	if switchNames.Current == switchNames.Original {
+		return nil
+	}
+
+	podAnnotation, err := util.UnmarshalPodAnnotation(pod.Annotations, "default")
+	if err != nil {
+		return fmt.Errorf("failed reading ovn annotation: %v", err)
+	}
+
+	node, err := oc.watchFactory.GetNode(pod.Spec.NodeName)
+	if err != nil {
+		return err
+	}
+	transitSwitchPortAddrs, err := util.ParseNodeTransitSwitchPortAddrs(node)
+	if err != nil {
+		return err
+	}
+	for _, podIP := range podAnnotation.IPs {
+		ipFamily := utilnet.IPFamilyOfCIDR(podIP)
+		transitSwitchPortAddr, err := util.MatchFirstIPNetFamily(ipFamily == utilnet.IPv6, transitSwitchPortAddrs)
 		if err != nil {
 			return err
 		}
+		route := nbdb.LogicalRouterStaticRoute{
+			IPPrefix: podIP.IP.String(),
+			Nexthop:  transitSwitchPortAddr.IP.String(),
+			Policy:   &nbdb.LogicalRouterStaticRoutePolicyDstIP,
+			ExternalIDs: map[string]string{
+				kubevirt.NamespaceExternalIDKey:    pod.Namespace,
+				kubevirt.OvnZoneExternalIDKey:      kubevirt.OvnRemoteZone,
+				kubevirtv1.VirtualMachineNameLabel: pod.Labels[kubevirtv1.VirtualMachineNameLabel],
+			},
+		}
+		if err := libovsdbops.CreateOrReplaceLogicalRouterStaticRouteWithPredicate(oc.nbClient, types.OVNClusterRouter, &route, func(item *nbdb.LogicalRouterStaticRoute) bool {
+			matches := item.IPPrefix == route.IPPrefix && item.Nexthop == route.Nexthop && item.Policy != nil && *item.Policy == *route.Policy
+			return matches
+		}); err != nil {
+			return fmt.Errorf("failed adding static route at remote zone: %v", err)
+		}
+	}
+	return nil
+}
 
-		if !isLiveMigrationLefover {
-			if err := oc.deleteDHCPOptions(pod); err != nil {
-				return err
+func (oc *DefaultNetworkController) ensureRoutingForVM(pod *kapi.Pod) error {
+	isLiveMigrationLefover, err := kubevirt.PodIsLiveMigrationLeftOver(oc.watchFactory, pod)
+	if err != nil {
+		return err
+	}
+	if util.PodWantsHostNetwork(pod) || !kubevirt.PodIsLiveMigratable(pod) || isLiveMigrationLefover {
+		return nil
+	}
+
+	targetNode := pod.Labels[kubevirtv1.NodeNameLabel]
+	targetReadyTimestamp := pod.Annotations[kubevirtv1.MigrationTargetReadyTimestamp]
+	// No live migration or target node was reached || qemu is already ready
+	if targetNode == pod.Spec.NodeName || targetReadyTimestamp != "" {
+		if oc.isPodScheduledinLocalZone(pod) {
+			if err := oc.ensureLocalZonePodAddressesToNodeRoute(pod); err != nil {
+				return fmt.Errorf("failed ensureLocalZonePodAddressesToNodeRoute for %s/%s: %w", pod.Namespace, pod.Name, err)
+			}
+		} else {
+			if err := oc.ensureRemoteZonePodAddressesToNodeRoute(pod); err != nil {
+				return fmt.Errorf("failed ensureRemoteZonePodAddressesToNodeRoute for %s/%s: %w", pod.Namespace, pod.Name, err)
 			}
 		}
 	}
 	return nil
 }
+
+func (oc *DefaultNetworkController) syncVirtualMachines(vms map[ktypes.NamespacedName]bool) error {
```

Comment:
should this be syncMigratedPods?

Reply:
It also involve not migrated VMs so VirtualMachines is what it is.